timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -461,66 +503,80 @@ def _train_multi(
-        fit_time_marginal_dict = self.get_models_attribute_dict("fit_time")
+        logger.debug("Generating leaderboard for all models trained")
 
-        for model_name in model_names:
-            score_val.append(score_dict[model_name])
-            fit_time_marginal.append(fit_time_marginal_dict[model_name])
+        model_names = self.get_model_names()
+        model_info = {}
+        for ix, model_name in enumerate(model_names):
+            model_info[model_name] = {
+                "model": model_name,
+                "fit_order": ix + 1,
+                "score_val": self.get_model_attribute(model_name, "val_score"),
+                "fit_time_marginal": self.get_model_attribute(model_name, "fit_time"),
+                "predict_time": self.get_model_attribute(model_name, "predict_time"),
+            }
 
-        test_score = []
         if data is not None:
-            logger.log(
-                30, "Additional data provided, testing on the additional data..."
+            logger.info(
+                "Additional data provided, testing on additional data. Resulting leaderboard "
+                "will be sorted according to test score (`score_test`)."
             )
             for model_name in model_names:
-                test_score.append(self.score(data, model_name))
-        df = pd.DataFrame(
-            data={
-                "model": model_names,
-                "score_val": score_val,
-                "fit_time_marginal": fit_time_marginal,
-                "fit_order": fit_order,
-            }
-        )
-        if test_score:
-            df["score_test"] = test_score
+                try:
+                    model_info[model_name]["score_test"] = self.score(data, model_name)
+                except Exception as e:  # noqa
+                    logger.error(
+                        f"Cannot score with model {model_name}. An error occurred: {str(e)}"
+                    )
+                    model_info[model_name]["score_test"] = float("nan")
 
-        df_sorted = df.sort_values(
-            by=["score_val", "model"], ascending=[False, False]
-        ).reset_index(drop=True)
+        df = pd.DataFrame(model_info.values())
 
-        df_columns_lst = df_sorted.columns.tolist()
-        explicit_order = ["model", "score_val", "fit_time_marginal", "fit_order"]
-        explicit_order = [
-            column for column in explicit_order if column in df_columns_lst
-        ]
-        df_columns_other = [
-            column for column in df_columns_lst if column not in explicit_order
+        sort_column = "score_test" if "score_test" in df.columns else "score_val"
+        df.sort_values(
+            by=[sort_column, "model"], ascending=[False, False], inplace=True
+        )
+        df.reset_index(drop=True, inplace=True)
+
+        explicit_column_order = [
+            "model",
+            "score_val",
+            "score_test",
+            "fit_time_marginal",
+            "predict_time",
```

Comment:
Reorder: predict_time is more important than fit_time as it is a performance indicator that is relevant during deployment whereas fit_time is not